lib/addressable/uri.rb

```diff
@@ -194,6 +194,8 @@ def self.heuristic_parse(uri, hints={})
         uri.sub!(/^file:\/+/i, "file:///")
       when /^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/
         uri.sub!(/^/, hints[:scheme] + "://")
+      else
+        uri = hints[:scheme] + "://" + uri if uri[/\A\d+\..*:\d+\z/]
```

Comment:
Why not just make this another when clause?

Reply:
sure, it's much better, thank you